### ISSUE TITLE
Focus lost when updating navigation link label

### ISSUE THREAD

**Issue by ajlende** [WordPress/gutenberg#46242]
### Description

On the navigation link and navigation submenu blocks, the label input loses focus when entering any text. 

### Step-by-step reproduction instructions

1. Select or add a navigation link to the navigation menu
2. In the block controls sidebar update the label
3. See that focus is lost and you can only type one or two letters at a time

### Screenshots, screen recording, code snippet

![focus lost when typing](https://user-images.githubusercontent.com/5129775/205098545-bfdba03e-5cf5-4910-b64c-cfae6485023e.gif)

### Environment info

- WordPress trunk (6.2.0)
- Gutenberg trunk (14.7.0-rc.1)
- Firefox 107.0

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by ajlende** [comment]
I was told this may have happened after [this refactor](https://github.com/WordPress/gutenberg/pull/46031). I haven't confirmed it myself, but it may be a place to start looking for the fix.

**Comment by getdave** [comment]
https://user-images.githubusercontent.com/444434/205139508-a89ccc0f-29be-440c-abf4-de0630c01ad2.mp4

Hmmmm I can't seem to replicate this in the Post Editor using Chrome. Any other reproduction steps?

**Comment by getdave** [comment]
Confirmed this is FF specific. I was able to replicate.

**Comment by getdave** [comment]
I think I have tracked this to `packages/block-editor/src/components/rich-text/use-firefox-compat.js`

...which is included on the `<RichText>` component here

https://github.com/WordPress/gutenberg/blob/d572cbacd12ade555193917f038fe819bdcfdb8c/packages/block-editor/src/components/rich-text/index.js#L443

If I remove the hook then I don't get the focus bug.

**Comment by getdave** [comment]
....but I don't know what the fix is. It is a bug with the hook or a bug in our implementation???

**Comment by MaggieCabrera** [comment]
It was not the refactor, but the bug is weirder than I initially thought. This is what I did:

I checked out https://github.com/WordPress/gutenberg/commit/821db1c8fc598a8dd69b52a97fe843f691b5cfa4 which is the commit previous to the refactor, and then added a link to my navigation and edited the label. It works fine for a few seconds and then the focus gets lost.

![ezgif-4-2e5e28f387](https://user-images.githubusercontent.com/3593343/205273451-b3036246-8cc5-46f7-8cf3-a762e362b387.gif)
